Expensify/App
# Add the `NewRoomPage`

### Details
This adds the New Room Page, which allows users to create a new room on a workspace.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/176850

### Tests
1. Click "New Room" in the global create menu. Verify it takes you to `<baseURL>/workspace/new-room`.
2. Verify the Room Name input functions correctly:
    - Type spaces, verify that they appear as underscores.
    - Try to type non-alphanumeric characters (besides underscores) into the room name input, verify nothing shows up.
    - Turn your caps-lock and type in the room name input, verify everything appears as lowercase.
2. Click the dropdown. Verify that all the workspaces (policies) shown match up to the "Free Plan" policies shown for the same account in OldDot (Settings > Policies)
3. Type `test_room_name` into the room name input and select a workspace. Click "Create Room". Verify that the room is created and that you navigate to the room (`<baseURL>/r/<reportID>`).
    - NOTE: The room name in the header, the room's entry in the LHN, and the default text won't show up correctly yet. Just verify that you are navigated to a new report. I will open up a separate PR for this, since it'll make this one too cluttered.
4. Navigate back to `<baseURL>/workspace/new-room`. Select the same workspace you used in [4]. Then, type `test_room_name` into the Room Name input. Verify that the `A room with this name already exists` error message displays (in real-time as you finish typing `test_room_name`).
5. Change the workspace in the workspace dropdown. Verify the error disappears. Reselect the same workspace, and verify the error appears again.
6. Delete the last `e` off of the room name. Verify the error disappears. Add the last `e` back, and verify the error appears again.

### QA Steps
Same as the above tests, just use these credentials on staging:
- username: `margrett952@labeledhf.com`
- password: `asdfASDF00`

### Tested On


### Screenshots


#### Web
https://user-images.githubusercontent.com/31285285/142488925-91679c72-7f4b-4109-986a-86eb8341d8a5.mp4

#### Mobile Web
<img width="375" alt="mweb" src="https://user-images.githubusercontent.com/31285285/142492811-2d357dd0-6c5d-4be3-831d-55513e48260d.png">

#### Desktop
https://user-images.githubusercontent.com/31285285/142492845-d9b47106-4409-48e9-b5c6-b17e671c3a2f.mp4

#### iOS
https://user-images.githubusercontent.com/31285285/142492814-78bf41c6-74ef-4b0e-a36f-e9e72300d4ca.mp4

#### Android
https://user-images.githubusercontent.com/31285285/142498036-bd255e0d-d1e9-478e-8bed-bad8fcb9f3d5.mp4
